sunxspex/thermal.py

```diff
@@ -0,0 +1,699 @@
+    line_flux = _line_emission(energy_edges_keV, temperature_K, abundance_type, relative_abundances)
+    flux = ((continuum_flux + line_flux) * emission_measure / 
+                (4 * np.pi * observer_distance**2))
+    if temperature.isscalar and emission_measure.isscalar:
+        flux = flux[0]
+    return flux
+
+
+@u.quantity_input(energy_edges=u.keV,
+                  temperature=u.K,
+                  emission_measure=(u.cm**(-3), u.cm**(-5)),
+                  observer_distance=u.cm)
+def continuum_emission(energy_edges,
+              temperature,
+              emission_measure,
+              abundance_type="sun_coronal",
+              relative_abundances=None,
+              observer_distance=(1*u.AU).to(u.cm)):
+    f"""Calculate the thermal X-ray continuum emission from the solar atmosphere.
+
+    The emission is calculated as a function of temperature and emission measure.
+    Which continuum mechanisms are included --- free-free, free-bound, or two-photon --- are
+    determined by the file from which the comtinuum parameters are loaded.
+    To change the file used, see the setup_continuum_parameters() function.
+
+    {doc_string_params}"""
+    temperature_K = temperature.to_value(u.K)
+    if temperature.isscalar:
+        temperature_K = np.array([temperature_K])
+    flux = _continuum_emission(energy_edges.to_value(u.keV), temperature_K,
+                               abundance_type=abundance_type,
+                               relative_abundances=relative_abundances)
+    flux *= emission_measure / (4 * np.pi * observer_distance**2)
+    if temperature.isscalar:
+        flux = flux[0]
+    return flux
+
+def _continuum_emission(energy_edges_keV,
+                        temperature_K,
+                        abundance_type,
+                        relative_abundances):
+    """
+    Calculates emission-measure-normalized X-ray continuum emission at the source.
+
+    Output must be multiplied by emission measure and divided by 4*pi*observer_distance**2
+    to get physical values.
+    Which continuum mechanisms are included --- free-free, free-bound, or two-photon --- are
+    determined by the file from which the comtinuum parameters are loaded.
+    To change the file used, see the setup_continuum_parameters() function.
+    
+    Parameters
+    ----------
+    energy_edges_keV: 1-D array-like
+        Boundaries of contiguous spectral bins in units on keV.
+
+    temperature_K: 1-D array-like
+        The temperature(s) of the plasma in unit of K.  Must not be a scalar.
+
+    abundance_type: `str`
+        Set of default abundances to be used.  See the continuum_emission function docstring
+        for options.
+
+    relative_abundances: same as for continuum_emission function.
+    """
+    # Handle inputs and derive some useful parameters from them
+    log10T_in = np.log10(temperature_K)
+    T_in_keV = temperature_K / 11604518  # Convert temperature from K to keV.
```

Comment:
Because it is way slower!  Also, this private function does not take quantities, but arrays, to the same reason.  So you would have to first convert `temperature_K` to a Quantity and then do the conversion, making it slower still.

Reply:
Still a bit of a magic number - could you define a module level variable `k_to_keV = (1 *u.k).to_value(u.keV)` that way more explicit?